pkg/aws/cluster/validate_cluster_test.go

```diff
@@ -25,31 +26,53 @@ func TestValidateReleaseVersion(t *testing.T) {
 			name: "case 0",
 			ctx:  context.Background(),
 
-			newReleaseVersion: unittest.DefaultReleaseVersion,
+			oldReleaseVersion: "3.0.0",
+			newReleaseVersion: "3.0.0",
 			valid:             true,
 		},
 		{
 			// version changed to valid release
 			name: "case 1",
 			ctx:  context.Background(),
 
-			newReleaseVersion: "1.0.0",
+			oldReleaseVersion: "3.0.0",
+			newReleaseVersion: "4.0.0",
 			valid:             true,
 		},
 		{
 			// version changed to deprecated release
 			name: "case 2",
 			ctx:  context.Background(),
 
-			newReleaseVersion: "2.0.0",
+			oldReleaseVersion: "3.0.0",
+			newReleaseVersion: "3.2.0",
 			valid:             false,
 		},
 		{
 			// version changed to invalid release
 			name: "case 3",
 			ctx:  context.Background(),
 
-			newReleaseVersion: "3.0.0",
+			oldReleaseVersion: "3.0.0",
+			newReleaseVersion: "3.3.0",
+			valid:             false,
+		},
+		{
+			// version changed with major skip
+			name: "case 4",
+			ctx:  context.Background(),
+
+			oldReleaseVersion: "3.0.0",
+			newReleaseVersion: "5.0.0",
```

Comment:
And now we would want to verify that skipping two or more is also covered. 🙂

Reply:
There are countless additional cases we could cover here. (what if we skip 2 releases? 3? 5?) However, I think it's good to keep the test cases basic and representative. If we find edge cases that we want to cover in the future, let"s add them here. However, this was not such a case since the upgrade validation is based on a simple `a greater-than b+1` condition that should remain true regardless of how many majors we skip.

Reply:
We do not have to cover everything, that is true. I for myself I just try to cover the obvious cases and since I saw some of them missing now after the tests became readable I suggested more cases. If you are fine with your test cases, you take the responsibility of ensuring this component does what it should even in two years down the road. I am ok with that. 🙂